mm2src/coins_activation/src/tendermint_with_assets_activation.rs

```diff
@@ -187,34 +189,45 @@ impl PlatformWithTokensActivationOps for TendermintCoin {
         })]
     }
 
-    async fn get_activation_result(&self) -> Result<Self::ActivationResult, MmError<Self::ActivationError>> {
+    async fn get_activation_result(
+        &self,
+        activation_request: &Self::ActivationRequest,
+    ) -> Result<Self::ActivationResult, MmError<Self::ActivationError>> {
         let current_block = self.current_block().compat().await.map_to_mm(|e| TendermintInitError {
             ticker: self.ticker().to_owned(),
             kind: TendermintInitErrorKind::RpcError(e),
         })?;
 
-        let balances = self.all_balances().await.mm_err(|e| TendermintInitError {
-            ticker: self.ticker().to_owned(),
-            kind: TendermintInitErrorKind::RpcError(e.to_string()),
-        })?;
+        let (balance, tokens_balances) = if activation_request.get_balances {
+            let balances = self.all_balances().await.mm_err(|e| TendermintInitError {
+                ticker: self.ticker().to_owned(),
+                kind: TendermintInitErrorKind::RpcError(e.to_string()),
+            })?;
+            (
+                CoinBalance {
+                    spendable: balances.platform_balance,
+                    unspendable: BigDecimal::default(),
+                },
+                balances
+                    .tokens_balances
+                    .into_iter()
+                    .map(|(ticker, balance)| {
+                        (ticker, CoinBalance {
+                            spendable: balance,
+                            unspendable: BigDecimal::default(),
+                        })
+                    })
+                    .collect(),
+            )
+        } else {
+            (CoinBalance::default(), HashMap::default())
```

Comment:
Because of how `enable_tendermint_with_assets` `ActivationResult` is set, I return balances as `0` if `get_balances` is `false`.

I didn't want to make the below 2 params optional until I get your opinion on this @ozkanonur https://github.com/KomodoPlatform/atomicDEX-API/blob/4bede306a9f4bc52a5121cf0f39bd60aa3a716f1/mm2src/coins_activation/src/tendermint_with_assets_activation.rs#L130-L131

Reply:
I think it makes more sense to make those balance fields `null` on the response for every coin type to have consistent response type which is also easier way to implement this endpoints for UI devs. So they will be 100% sure if these fields were `null`, they didnt enable the balance on requests.